vault/identity_store_oidc_provider.go

```diff
@@ -48,6 +48,14 @@ type provider struct {
 	effectiveIssuer string
 }
 
+type providerDiscovery struct {
+	discovery
```

Comment:
I am not sure if embedding this struct is desirable. It makes for a bit of an odd interface to the struct. The other option would be to simply copy over the `discovery` fields.

Reply:
I'm okay with either approach, but I think I'd prefer to copy over the fields. We should also add the recommended `scopes_supported` field, since we have that data for each provider (ref: [ProviderMetadata](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata)). We'll want to make sure to include the `openid` scope for every provider + configured scopes.

Reply:
Thanks! Totally forgot about scopes_supported